src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -506,6 +491,61 @@ private static StringBuilder fakeBody(final Request... dispositions)
                 .append(String.format("--%s--", BOUNDARY));
             return builder;
         }
+
+        /**
+         * This class is  for the specific implementation
+         * for the RqMultipart. see the overridden methods for details.
+         *
+         * @todo #517:30min re-design the class.
+         *  Why the super class is encapsulated inside the class?
```

Comment:
@mkordas we can get the header of the current object. Then why we need an encapsulated Request?
If this is a class to do a one to many header to body mapping it doesn't need to be of the type of Request.
In my view :)